generator/internal/format_method_comments.cc

```diff
@@ -258,32 +266,60 @@ bool CheckMethodCommentSubstitutions() {
   return all_substitutions_used;
 }
 
-std::string FormatStartMethodComments() {
-  return R"""(  // clang-format off
-  ///
+std::string FormatStartMethodComments(bool is_method_deprecated) {
+  auto constexpr kMethodCommentsPrefix = R"""(  // clang-format off
+  ///)""";
+
+  auto constexpr kMethodCommentsSuffix = R"""(  ///
+  // clang-format on
+)""";
+
+  auto constexpr kCommentBody = R"""(
   /// @copybrief $method_name$
   ///
   /// Specifying the [`NoAwaitTag`] immediately returns the
   /// [`$longrunning_operation_type$`] that corresponds to the Long Running
   /// Operation that has been started. No polling for operation status occurs.
   ///
   /// [`NoAwaitTag`]: @ref google::cloud::NoAwaitTag
+)""";
+
+  std::string const deprecation_comment =
+      is_method_deprecated ?
+                           R"""( @deprecated This RPC is deprecated.
   ///
+  ///)"""
+                           : "";
+
+  return absl::StrCat(kMethodCommentsPrefix, deprecation_comment, kCommentBody,
+                      kMethodCommentsSuffix);
+}
+
+std::string FormatAwaitMethodComments(bool is_method_deprecated) {
+  auto constexpr kMethodCommentsPrefix = R"""(  // clang-format off
```

Comment:
It seems like there is an opportunity to define `kMethodCommentsPrefix`, `kMethodCommentsSuffix`, `kDeprecatedComment` once.

Reply:
Done.